### ISSUE TITLE
Proposal: JUDGE+

### ISSUE THREAD

**Issue by hadronyche** [seraxis/lr2oraja-endlessdream#8]
A trainer setting to target a given judgerank. The target will be applied if it's harder than the loaded chart's judgerank.

E.g. if I want to practice normal judge and set the target to normal judgerank: all easy judge charts will load as normal judge.

**(1)** Beatoraja currently lets you proxy this via `EXPAND JUDGE` and does not class it as an assist if all values are <=100. However, while it's simple to toggle on and off in-game, it's tedious to calculate the numbers (take from iidx.org) and input the numbers, and have to go into the launcher when you want to change them. A simple drop-down box to select judgeranks will be much more convenient.

Careful consideration should be given to how and whether this should affect charthash. Changing the judgerank in a `bms` file does affect the charthash. However, since the behaviour described in **(1)** is allowed without any repercussions (i.e. `ASSIST`, charthash change), there is a case to be made for it to not **(A)**. On the other hand, changing the chart hash will allow a separate tracking of scores on that judgerank **(B)**. Scores on higher judgeranks are difficult to view outside of result, but can be shown with a message on chart load (e.g. `EASY CLEAR, 4111 EX, 91.02% AAA`)

I've not investigated how possible behaviours **(A)** or **(B)** are to implement, but I would appreciate thoughts on which you think would be better to build for. My guess is that it's simple to implement and changing charthash dynamically should naively store the score in the DB, though there may be a conflict wherein a chart path with multiple charthashes may have undesired behaviour.

**Comment by hadronyche** [comment]
Agreed, saving the local score as NO PLAY seems like the most simple, elegant solution and leaves extensibility open down the line.

An additional consideration is that most IRs support manual submission (LR2oraja: Tachi, CinnamonIR), which would send scores previously limited to local DB saves. This currently affects the Rate Modifier (to my knowledge, untested), which raises a discussion in itself about how much to cater to IRs and how this would limit future features. The use of FREQ in LR2 and tighter judges by EXPAND JUDGE in raja are obvious and allowed to submit on their IRs, but novel features without any precedent or parallel will require discussion.

I haven't investigated this but applying something like LIGHT ASSIST may allow IRs to ignore these scores. Otherwise, your idea of a separate DB would avoid this issue or we could bring features in line with what IRs accept to send scores.

**Comment by Catizard** [comment]
I agree we should treat the scores that is earned under stricter window differently because it's more like an achievement, and different judge rank scores don't have much value to compare. As describe before, (A) would pollute the database file and there's no way back; (B) is a great idea since we can keep the drop-in compatibility easily. However, as described in the connected pr, it would be very hard to implement and tricky to handle what scores we present for the user. 

So I think we should rise another proposal for setting up a database file for endless dream only. And we can move the discussion there since both freq and this feature should be considered under that proposal.

Currently the connected pr doesn't treat the scores specially, here's the summary:
- If the overriding judge rank is the same or stricter than original one, nothing changes. The scores would be updated when it's better, lost if not(beatoraja doesn't store your every play log correctly, so what I mean lost it's really lost)
- Otherwise, you are playing the chart under a wider judge window, your clear cannot be better than `L-Assist` and scores won't be updated or sent to ir. This is the same behavior when you setting judge window > 100 in "expand judge".
